massa-execution-worker/src/execution.rs

```diff
@@ -616,6 +620,59 @@ impl ExecutionState {
         Ok(())
     }
 
+    pub fn execute_denunciation_op(
+        &self,
+        operation: &OperationType,
+        _sender_addr: Address,
+        block_credits: &mut Amount,
+    ) -> Result<(), ExecutionError> {
+
+        const ROLL_COUNT: u64 = 1;
+
+        // process denunciation operations only
+        let denunciation = match operation {
+            OperationType::Denunciation { data } => data,
+            _ => panic!("unexpected operation type"),
+        };
+
+        if denunciation.is_valid() {
```

Comment:
the denunciation validity should be checked in the context of the block: if it refers to an event that is "too old" it is invalid